### PR TITLE
feat(TFD-1666): Use JsonSchemaInferrer & JsonGenericRecordConverter

### PR DESCRIPTION
**Pull request by zafkirTalend**
**What is the current behavior?** (You can also link to an open issue here)

https://jira.talendforge.org/browse/TFD-1666

**What is the new behavior?**

Map the PythonRow output schema into an avro schema.

**Please check if the PR fulfills these requirements**

- [ ] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [x] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

**Other information**:
https://jira.talendforge.org/browse/TFD-1666

### CODE REVIEW

File: components/components-processing/processing-runtime/src/main/java/org/talend/components/processing/runtime/pythonrow/PythonRowDoFn.java [hunk 1]
```diff
@@ -105,10 +105,11 @@ private void map(IndexedRecord input, ProcessContext context) throws IOException
         python.exec("outputJSON = json.dumps(output)");
         PyObject output = python.get("outputJSON");
 
-        ByteArrayInputStream bais = new ByteArrayInputStream(output.toString().getBytes());
-        DataInputStream din = new DataInputStream(bais);
-        JsonDecoder decoder = DecoderFactory.get().jsonDecoder(schema, din);
-        context.output(reader.read(null, decoder));
+        JsonSchemaInferrer jsonSchemaInferrer = new JsonSchemaInferrer(new ObjectMapper());
```
Comment (pbailly) [inline components/components-processing/processing-runtime/src/main/java/org/talend/components/processing/runtime/pythonrow/PythonRowDoFn.java]: The line 108-110 are going to be a heavy processing.
Could you use a cache mechanism on the jsonGenericRecordConverter?

File: components/components-processing/processing-runtime/src/main/java/org/talend/components/processing/runtime/pythonrow/PythonRowDoFn.java [hunk 2]
```diff
@@ -127,10 +128,11 @@ private void flatMap(IndexedRecord input, ProcessContext context) throws IOExcep
         if (outputList instanceof PyList) {
             PyList list = (PyList) outputList;
             for (Object output : list) {
-                ByteArrayInputStream bais = new ByteArrayInputStream(output.toString().getBytes());
-                DataInputStream din = new DataInputStream(bais);
-                JsonDecoder decoder = DecoderFactory.get().jsonDecoder(schema, din);
-                context.output(reader.read(null, decoder));
+                JsonSchemaInferrer jsonSchemaInferrer = new JsonSchemaInferrer(new ObjectMapper());
```
Comment (pbailly) [inline components/components-processing/processing-runtime/src/main/java/org/talend/components/processing/runtime/pythonrow/PythonRowDoFn.java]: Could you also use a cache mechanism here?

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/819/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/819/)

File: components/components-processing/processing-definition/src/main/java/org/talend/components/processing/pythonrow/PythonRowProperties.java [hunk 3]
```diff
@@ -40,8 +40,6 @@ public PythonRowProperties(String name) {
 
     public Property<MapType> mapType = PropertyFactory.newEnum("mapType", MapType.class);
 
-    public Property<Boolean> changeOutputSchema = PropertyFactory.newBoolean("changeOutputSchema");
-
     // TODO: Currently the user have to set the schema property if he want to specify the output schema.
```
Comment (pbailly) [inline components/components-processing/processing-definition/src/main/java/org/talend/components/processing/pythonrow/PythonRowProperties.java]: Remove this comments.

File: components/components-processing/processing-definition/src/main/java/org/talend/components/processing/pythonrow/PythonRowProperties.java [hunk 4]
```diff
@@ -40,8 +40,6 @@ public PythonRowProperties(String name) {
...
     // This has high risk to be changed in short time.
     public SchemaProperties schemaFlow = new SchemaProperties("schemaFlow");
```
Comment (pbailly) [inline components/components-processing/processing-definition/src/main/java/org/talend/components/processing/pythonrow/PythonRowProperties.java]: SchemaFlow is unecessary, remove it too.

File: components/components-processing/processing-runtime/src/main/java/org/talend/components/processing/runtime/pythonrow/PythonRowDoFn.java [hunk 5]
```diff
@@ -69,7 +71,7 @@ public void processElement(ProcessContext context) throws IOException {
             IndexedRecord input = (IndexedRecord) converter.convertToAvro(context.element());
 
             if (schema == null) {
-                schema = retrieveOutputSchema(input);
+                schema = input.getSchema();
             }
             if (reader == null) {
                 reader = new GenericDatumReader<>(schema);
```
Comment (pbailly) [inline components/components-processing/processing-runtime/src/main/java/org/talend/components/processing/runtime/pythonrow/PythonRowDoFn.java]: This variable "reader" is no longer used. 
Do the same with the "schema".

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/821/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/821/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/822/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/822/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/839/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/839/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/866/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/866/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/867/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/867/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/882/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/882/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/930/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/930/)

**Pull request merged**